idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,651 @@
+                    continue;
+                }
+                String transitiveClosureKey = str.substring(0, str.indexOf(","));
+                String transitiveClosureValues = str.substring(str.indexOf(","));
+                String[] transitiveClosureValueArray = transitiveClosureValues.split(",");
+                Set<String> transitiveClosureValue = new HashSet<>();
+                for (String transitiveClosureValueArrayItem: transitiveClosureValueArray) {
+                    transitiveClosureValue.add(transitiveClosureValueArrayItem);
+                }
+                reverseTransitiveClosure.put(transitiveClosureKey, transitiveClosureValue);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        // the dependency map from classes to their dependencies
+        Map<String, Set<String>> transitiveClosure = getReverseClosure(reverseTransitiveClosure);
+
+        Set<String> additionalTests = new HashSet<>();
+
+        // iter through the affected tests and find what depends on
+        Set<String> processedClasses = new HashSet<>();
+
+        getImmutableList();
+        for (String affectedTest : affectedTests) {
+            Set<String> dependencies = transitiveClosure.get(affectedTest);
+            if (dependencies == null) {
+                continue;
+            }
+            for (String dependency : dependencies) {
+                if (processedClasses.contains(dependency)) {
+                    continue;
+                }
+                try {
+                    Class clazz = loader.loadClass(dependency);
+                    for (Field field : clazz.getDeclaredFields()) {
+                        if (inImmutableList(field) && Modifier.isFinal(field.getModifiers())) {
+                            continue;
+                        }
+                        if (Modifier.isStatic(field.getModifiers())) {
+                            String upperLevelAffectedClass = clazz.getName();
+                            Set<String> upperLevelAffectedTestClasses = reverseTransitiveClosure.get(upperLevelAffectedClass);
+                            if (upperLevelAffectedTestClasses != null) {
+                                for (String upperLevelAffectedTestClass: upperLevelAffectedTestClasses) {
+                                    additionalTests.add(upperLevelAffectedTestClass);
+                                }
+                            }
+                            break;
+                        }
+                    }
+                    processedClasses.add(dependency);
+                } catch (ClassNotFoundException | NoClassDefFoundError e)  {
+                    e.printStackTrace();
+                }
+
+            }
+        }
+
+        affectedTests.addAll(additionalTests);
+        return affectedTests;
+    }
+
+    private Set<String> computeEkstaziAffectedTests(final MavenProject project) throws IOException, MojoExecutionException {
```

Comment:
Yeah, this method looks very similar to computeAffectedTests, should just strip out what actually is different between Ekstazi and STARTS, get the selected tests and dependencies into some consistent internal format, then use the same core logic for the analysis.